compiler-core/src/erlang/tests.rs

```diff
@@ -658,3 +658,19 @@ pub fn main() {
 "
     );
 }
+
+// https://github.com/gleam-lang/gleam/issues/3382
+#[test]
+fn function_named_module_info() {
+    assert_erl!(
+        "
+pub type ModuleInfo {
+    ModuleInfo
+}
+
+pub fn module_info() {
+    ModuleInfo
+}
+"
```

Comment:
Can we get a test case that shows the function being called, it being used as a variable, it being used as a variable in a constant, all of the above while imported and qualified, and imported and unqualified. Thank you

Reply:
I added all permutations I could think of, please let me know if I missed any. And also I am not sure if I implemented the tests in a way I should (didn't see any similar present)

Reply:
All the ones you've got for constants are needed for functions, as well as referencing a `module_info` function in a constant and then using that constant.